pkgs/applications/misc/ilspy/default.nix

```
@@ -0,0 +1,99 @@
+{ lib
+, stdenv
+, pkgs
+
+, fetchzip
+, unzip
+, autoPatchelfHook
+, makeWrapper
+, makeDesktopItem
+
+#, stdenv.cc.cc.lib
+, lttng-ust
+, libkrb5
+, zlib
+, fontconfig
+
+, openssl_1_1
+, xorg
+, libX11
+, libICE
+, libSM
+, icu
+}:
+stdenv.mkDerivation rec {
+  pname = "ILSpy";
```

```suggestion
  pname = "AvaloniaILSpy";
```

I would put it lower but I'll leave that to you

---

`Avalonia` refers to the UI system employed on the base project ILSpy and is prefixed only to differentiate on github. Aside from the Windows build no distribution exists employing any other UI system. Hence not carrying that prefix over.

---

we can conceivably package ilspy and avaloniailspy there is no reason to avoid keeping the prefix

---

Should the windows version of this project be packaged for cross-platform execution, then that package should replace this one.

---

It's also important for people to clearly know which edition of ILSpy they're using so issues are reported to the right GH repo

https://github.com/icsharpcode/ILSpy/issues/2274#issuecomment-760907411

---

Hm. Scratch that. The URLs _do_ already point to the right project. Back to not seeing the idea here.

---

> Don't see the sense in the pname change though.

right now it suggestes that this is ILSpy. IIRC it was in nixpkgs at some point or there was open PR. A fork should be clearly distinguishable.

---

To be clear, if for example all gtk-based application packages had a `gtk` prefix then the point would be obvious and users would know to look for the package `gtk4-pika-backup` rather than `pika-backup` - in case someone might package `qt5-pika-backup` at some point.